workers/unity/Packages/io.improbable.gdk.debug/WorkerInspector/EntityList.cs

```diff
@@ -0,0 +1,268 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Profiling;
+using UnityEditor;
+using UnityEditor.UIElements;
+using UnityEngine.UIElements;
+
+namespace Improbable.Gdk.Debug.WorkerInspector
+{
+    public class EntityList : VisualElement
+    {
+        public delegate void EntitySelected(EntityId entityId);
+
+        public EntitySelected OnEntitySelected;
+
+        private World world;
+        private EntitySystem entitySystem;
+
+        private readonly ListView listView;
+
+        private readonly Entities entities = new Entities();
+
+        private int lastViewVersion;
+
+        public EntityList()
+        {
+            const string uxmlPath = "Packages/io.improbable.gdk.debug/WorkerInspector/Templates/EntityList.uxml";
+            var template = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>(uxmlPath);
+            template.CloneTree(this);
+
+            listView = this.Q<ListView>();
+            listView.itemHeight = 24;
+            listView.makeItem = () => new Label();
+            listView.bindItem = BindItem;
+            listView.onSelectionChanged += OnSelectionChanged;
+            listView.itemsSource = entities.Data;
+
+            var searchField = this.Q<ToolbarSearchField>();
+            searchField.RegisterCallback<ChangeEvent<string>>(OnSearchFieldChanged);
+        }
+
+        public void Update()
+        {
+            if (entitySystem == null)
+            {
+                listView.itemsSource = new List<EntityId>();
+                return;
+            }
+
+            if (entitySystem.ViewVersion != lastViewVersion)
+            {
+                entities.RefreshData(world);
+                listView.itemsSource = entities.Data;
+                lastViewVersion = entitySystem.ViewVersion;
+            }
+        }
+
+        public void SetWorld(World world)
+        {
+            this.world = world;
+            entitySystem = world?.GetExistingSystem<EntitySystem>();
+            lastViewVersion = 0;
+        }
+
+        private void BindItem(VisualElement element, int index)
+        {
+            var entity = entities.Data[index];
+
+            if (!(element is Label label))
+            {
+                return;
+            }
+
+            label.text = entity.ToString();
+        }
+
+        private void OnSelectionChanged(List<object> selections)
+        {
+            // TODO: Error handling
```

Comment:
Reminder to put in error handling. :)

Reply:
See https://github.com/spatialos/gdk-for-unity/pull/1375/commits/e3d3542306b6820939f1188eac77dd95eefdaae4 :)